components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -2185,6 +2206,43 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_cast_year_as_time() {
+        let normal_cases = vec![
+            ("2020-00-00 00:00:00", 2020),
+            ("2000-00-00 00:00:00", 2000),
+            ("1999-00-00 00:00:00", 1999),
+            ("2077-00-00 00:00:00", 2077),
+            ("1901-00-00 00:00:00", 1901),
+            ("2155-00-00 00:00:00", 2155),
+            ("0000-00-00 00:00:00", 0),
```

Comment:
Myabe we should add test case like `1900` and `2156`?

Reply:
done